code/game/turfs/turf.dm

```diff
@@ -181,29 +181,117 @@
 			return 0
 	return 1 //Nothing found to block so return success!
 
-/turf/Entered(atom/movable/Obj,atom/OldLoc)
+/turf/Entered(atom/movable/A as mob|obj)
+	if(movement_disabled)
+		to_chat(usr, "<span class='warning'>Movement is admin-disabled.</span>")//This is to identify lag problems
+		return
+	//THIS IS OLD TURF ENTERED CODE
 	var/loopsanity = 100
-
-	if(ismob(Obj))
-		if(Obj.areaMaster && Obj.areaMaster.has_gravity == 0)
-			inertial_drift(Obj)
+	if(ismob(A))
+		if(A.areaMaster && A.areaMaster.has_gravity == 0)
+			inertial_drift(A)
 	/*
-		if(Obj.flags & NOGRAV)
-			inertial_drift(Obj)
+		if(A.flags & NOGRAV)
+			inertial_drift(A)
 	*/
 
 		else if(!istype(src, /turf/space))
-			Obj:inertia_dir = 0
+			A:inertia_dir = 0
 	..()
 	var/objects = 0
-	if(Obj && Obj.flags & PROXMOVE)
-		for(var/atom/A as mob|obj|turf|area in range(1))
+	if(A && A.flags & PROXMOVE)
+		for(var/atom/Obj as mob|obj|turf|area in range(1))
 			if(objects > loopsanity)	break
 			objects++
 			spawn( 0 )
-				if ((A && Obj) && A.flags & PROXMOVE)
-					A.HasProximity(Obj, 1)
-	return
+				if ((A && Obj) && Obj.flags & PROXMOVE)
+					Obj.HasProximity(A, 1)
+	// THIS IS NOW TRANSIT STUFF
```

Comment:
What are there any checks that prevent you from building on the map edge? If so: nevermind.